### PR TITLE
Avoid ragged arrays when unwrapping nested parameters

### PR DESCRIPTION
**Pull request by albi3ro**
Contains just part of #3021 , but that PR was also going to fix batching for operator arithmetic.  Operator arithmetic batching will take more PR's and thought.

The following code:
```
import warnings
warnings.simplefilter("error")

with qml.tape.QuantumTape() as tape:
    obs = qml.op_sum(qml.s_prod(torch.tensor(1.2), qml.PauliX(0)), qml.PauliZ(0))
    qml.expval(obs)

with qml.tape.Unwrap(tape):
    print(tape.get_parameters())
    print(obs.parameters)
```
Will still raise errors, but that will be because of batching and not because of unwrapping.  So there will be one less warning.

The warning raised by unwrapping the nested data will no longer exist.

### CODE REVIEW

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -85,6 +85,8 @@ keyword argument when using `GellMann`, which determines which of the 8 Gell-Man
 * Add details to the output of `Exp.label()`.
   [(#3126)](https://github.com/PennyLaneAI/pennylane/pull/3126)
 
+* `qml.math.unwrap` no longer creates ragged arrays. Lists remain lists.
```
Comment (albi3ro) [inline doc/releases/changelog-dev.md:88]:
```suggestion
* `qml.math.unwrap` no longer creates ragged arrays. Lists remain lists.
  [(#3163)](https://github.com/PennyLaneAI/pennylane/pull/3163)
```

File: pennylane/math/multi_dispatch.py [hunk 2]
```diff
@@ -742,21 +742,21 @@ def unwrap(values, max_depth=None):
...
+        new_val = (
+            np.to_numpy(val, max_depth=max_depth) if isinstance(val, ArrayBox) else np.to_numpy(val)
+        )
 
-        if isinstance(a, ndarray) and not a.shape:
-            # if NumPy array is scalar, convert to a Python float
-            res.append(a.tolist())
-        else:
-            res.append(a)
+        if isinstance(new_val, ndarray) and not new_val.shape:
```
Comment (AlbertMitjans) [inline pennylane/math/multi_dispatch.py:753]: Isn't `new_val` always a `ndarray`?

**Review by AlbertMitjans — COMMENTED** [review]
Just one small comment

File: pennylane/math/multi_dispatch.py (hunk 2 above)
Comment (albi3ro) [reply pennylane/math/multi_dispatch.py:753]:
```suggestion
        if not new_val.shape:
```
Good point.  I was just modifying the existing code and not thinking too much about it.  Let's see if this breaks it.

Comment (albi3ro) [reply pennylane/math/multi_dispatch.py]: It won't be an `ndarray` if `new_val` is a sparse matrix.

**Review by AlbertMitjans — APPROVED** [review]
Approved!

**Pull request merged**